Applications/utils/npy_reader/npy_reader.cpp

```diff
@@ -0,0 +1,119 @@
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * Copyright (C) 2024 Donghak Park <donghak.park@samsung.com>
+ *
+ * @file   npy_reader.cpp
+ * @date   28 Feb 2024
+ * @brief  reader for npy file
+ * @see    https://github.com/nnstreamer/nntrainer
+ * @author Donghak Park <donghak.park@samsung.com>
+ * @bug    No known bugs except for NYI items
+ */
+
+#include "npy_reader.h"
+
+#include <assert.h>
+#include <cstring>
+#include <iostream>
+#include <random>
+#include <string>
+#include <vector>
+
+namespace nntrainer::util {
+namespace {
+/**
+ * @brief read numpy file from file_path
+ *
+ * @param file_path file_path that want to read
+ * @param dims data shape
+ * @param values return value that contain npy's contents
+ */
+void read_npy_file(const char *file_path, std::vector<int> &dims,
+                   std::vector<float> &values) {
+  FILE *file = fopen(file_path, "rb");
+
+  char magic[7] = {};
+  fread(magic, 6, 1, file);
+  assert(strcmp(magic, "\x93NUMPY") == 0);
```

Comment:
assert is not a good idea for a library if it fails because of invalid parameters from the app. return a proper error number or throw an exception.

this function also needs proper error handling (e.g., invalid fail_path) unless this is for test purpose only.

Reply:
1. I replaced the assert statement with try catch and ml_error.
2. I added error checking for file operations.
3. make if statement that check fread count